presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java

```diff
@@ -2395,6 +2395,17 @@ public void testShowCatalogsLike()
         assertEquals(result.getOnlyColumnAsSet(), ImmutableSet.of(getSession().getCatalog().get()));
     }
 
+    @Test
+    public void testShowCatalogsLikeWithEscape()
+    {
+        assertQueryFails("SHOW CATALOGS LIKE '%$_%' ESCAPE", "line 1:33: mismatched input '<EOF>'. Expecting: <string>");
+        assertQueryFails("SHOW CATALOGS LIKE 't$_%' ESCAPE ''", "Escape string must be a single character");
+        assertQueryFails("SHOW CATALOGS LIKE 't$_%' ESCAPE '$$'", "Escape string must be a single character");
+
+        MaterializedResult result = computeActual("SHOW CATALOGS LIKE '$_%' ESCAPE '$'");
```

Comment:
Move this one to an engine-specific test (similar to the ones in `io.prestosql.sql.query.TestXXX`). It'd be nice to also have a positive test. There's a catalog named `testing_catalog` in LocalQueryRunner that you can use for that purpose.